cli/src/main/java/com/devonfw/tools/ide/commandlet/UpgradeSettingsCommandlet.java

```diff
@@ -181,4 +187,43 @@ private static void updatePropertiesLegacyEdition(EnvironmentVariablesProperties
       environmentVariables.remove(legacyEditionVariable);
     }
   }
+
+  private void cleanupLegacyProperties() {
+    this.context.info("Cleaning up legacy properties...");
+
+    Path rootDirectory = Paths.get(System.getProperty("user.dir"));
+    try {
+      Files.walk(rootDirectory)
+          .filter(path -> path.getFileName().toString().equals("IDEasy.properties"))
+          .forEach(filePath -> {
+            try {
+              updateProperties(filePath);
+            } catch (IOException e) {
+              this.context.warning("Error processing IDEasy.properties at " + filePath);
+            }
+          });
+    } catch (IOException e) {
+      this.context.warning("Error walking the file tree to find IDEasy.properties.");
+    }
+  }
+
+  private void updateProperties(Path filePath) throws IOException {
+    List<String> lines = Files.readAllLines(filePath);
+    List<String> updatedLines = new ArrayList<>();
+
+    for (String line : lines) {
+      if (line.startsWith("git.url=") || line.startsWith("git-url")) {
+        String gitUrl = line.substring("git.url=".length());
+        updatedLines.add("git_url=" + gitUrl);
+        continue;
+      }
+      if (line.startsWith("eclipse=import")) {
+        updatedLines.add("import=eclipse");
+        continue;
+      }
+      updatedLines.add(line);
+    }
+    Files.write(filePath, updatedLines, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING);
```

Comment:
you should add a boolean flag `boolean updated=false;` before the loop and set it to `true` when you have modified something.
Then you can only write the file changes back if something changed and additionally log the success message only in that case so if nothing changed then nothing if logged.